core/uemis-downloader.c

```diff
@@ -1200,7 +1206,7 @@ static bool get_matching_dive(int idx, char *newmax, int *uemis_mem_status, devi
 						fprintf(debugfile, "TRY matching dive log id %d from %s with dive details %d but details are deleted\n", dive->dc.diveid, d_time, dive_to_read);
 #endif
 						deleted_files++;
-						max_deleted_seen = dive_to_read;
+						max_deleted_seen = dive->number;
```

Comment:
This change could use some explanation... the internal numbers of the Uemis are completely independent from our dive numbers. I'm not sure I understand why this is the right thing to do...

Reply:
Actually, this is the only line that is also bothering me. I suspect that max_deleted_seen has no effect at all because it is only read in function uemis_get_divenr and this function is called only once in do_uemis_import - and that is before the loop which contains get_matching_dive and thus writes max_deleted_seen.
If I am right, we should remove the variable max_deleted_seen completely.

Besides that, I am on your side: dive_to_read addresses the uemis' internal number - not the dive(->)number. However, max_deleted_seen is compared to the dive id (or dive number, respectively) in uemis_get_divenr. So, it makes more sense to me to update it using dive_number.

Reply:
So kann we split this in two commits?
One without this change.
And another commit, that either removes the variable (if it indeed isn't needed), or changes things to be consistent (and documents that in the commit message).
Thanks

Reply:
I just made some tests with the Uemis: uemis_get_divenr is only called once during the whole process. I removed max_deleted_seen and it did not make any difference. I will revert the change at line 1209 in this PR and create a new one to remove max_deleted_seen. Thank you.